crates/bevy_color/src/color_range.rs

```diff
@@ -0,0 +1,63 @@
+use std::ops::Range;
+
+use crate::{LinearRgba, Mix};
+
+/// Represents a range of colors that can be linearly interpolated, defined by a start and
+/// end point which must be in the same color space. It works for any color type that
+/// implements [`Mix`].
+///
+/// This is useful for defining gradients or animated color transitions.
+pub trait ColorRange<T: Mix> {
+    /// Get the color value at the given interpolation factor, which should be between 0.0 (start)
+    /// and 1.0 (end).
+    fn at(&self, factor: f32) -> T;
+}
+
+impl<T: Mix> ColorRange<T> for Range<T> {
+    fn at(&self, factor: f32) -> T {
+        self.start.mix(&self.end, factor)
+    }
+}
+
+/// A type-erased color range that can be used to interpolate between colors in various
+/// color spaces. Note that both the start and end points must be in the same color space.
+/// This is useful for defining an animated color transition, such that the color space can
+/// be chosen when the range is created, but the interpolation can be done without knowing
+/// the color space.
+pub trait AnyColorRange {
```

Comment:
I'm still having difficulty piecing together the usage of this type from the definition and docs alone. Can you add a doc test that shows how this trait could be used for interpolating for e.g. a gradient, and why this trait is needed?

Reply:
I can also just remove this. I use it in my own UI work, but I can just as easily move the trait definition to my UI crate.

The way I use this is that I have an ECS component that animates a UI property: examples are `AnimatedBackgroundColor`, `AnimatedBorderColor`, `AnimatedOutlineColor` and so on. Within the component there's a color range that specifies the beginning and ending of the transition. You can think of the start and end as keyframes in an animation sequence, except that there's only two.

The `AnyColorRange` trait lets me interpolate in different color spaces without having to have different component types, i.e. `AnimatedBackgroundColorRgb`, `AnimatedBackgroundColorHsl` and so on. Because it's a dyn trait, the color space type is erased.

The problem with adding a doc test is that the use case is rather elaborate, and it would make for a fairly verbose test.

Reply:
Yeah, let's cut this for now: we can always add it back in once we have a tangible first-party use case.